[C#] Simplified basic-enums exercise by removing parsing (#864)
exercism/v3: languages/csharp/exercises/concept/enums/.meta/design.md

- Know how to define enum members.
- Know how to assign values to enum members.
- Know how to get an enum's numeric value.
- Know how to convert an `enum` to a `string`

## Out of scope

- Flag enums.
- That an enum's underlying type can be changed.
- Memory and performance characteristics.
- Parse `string` into an `enum`.

## Concepts

The Concepts this exercise unlocks are:

> - `enums-basic`: know of the existence of the `enum` keyword; know how to define enum members; know how to assign values to enum members; know how to get an enum's numeric value; know how to convert a `string` to an `enum` and vice versa.

Reviewer: ```suggestion
- `enums`: know of the existence of the `enum` keyword; know how to define enum members; know how to assign values to enum members; know how to get an enum's numeric value; know how to convert an `enum` to a `string`.
```

> - `conditionals-switch-basic`: know how to use a `switch` statement using constant values.

Reviewer: ```suggestion
- `constant-switch`: know how to use a `switch` statement using constant values.
```

## Prequisites

This exercise's prerequisites Concepts are:

> - `strings-basic`: log lines are `string` values.

Reviewer: ```suggestion
- `strings`: log lines are `string` values.
```

> - `numbers-basic`: enum values are assigned a numeric value.
> - `conditionals-if`: used to handle the case when the enum could not be parsed.
> - `type-conversion-numbers`: casting the enum to an `int`.

Reviewer: ```suggestion
- `conditionals`: know how to execute conditional logic.
```

- Know how to define enum members.
- Know how to assign values to enum members.
- Know how to get an enum's numeric value.

> - Know how to convert an `enum` to a `string`

Reviewer: This is not exactly what we want students to learn, but instead:

```suggestion
- Know how to use a `switch` statement to handle multiple conditions.
```

Reviewer: @batibot323 Could you add the above suggestion?

## Out of scope

- Flag enums.
- That an enum's underlying type can be changed.
- Memory and performance characteristics.

> - Parse `string` into an `enum`

Reviewer: This is correct! Good addition.

```suggestion
- Parse `string` into an `enum`.
```

## Concepts

The Concepts this exercise unlocks are:

- `enums-basic`: know of the existence of the `enum` keyword; know how to define enum members; know how to assign values to enum members; know how to get an enum's numeric value; know how to convert a `string` to an `enum` and vice versa.

> - `switch-basic`: know how to use a simple switch-case statement.

Reviewer: ```suggestion
- `conditionals-switch-basic`: know how to use a `switch` statement using constant values.
```

> - `strings-conversion`: know how to convert from an enum to string. [not if this is already a Concept or should just belong to strings-basic]

Reviewer: This is already covered in `enums-basic`.

```suggestion
```

## Prequisites

This exercise's prerequisites Concepts are:

- `strings-basic`: log lines are `string` values.

> - `numbers-basic`: enum values are assigned a numeric value.

Reviewer: Good to see this removed :)
